students/morgan/Mailroom/mailroom4.py

```diff
@@ -0,0 +1,215 @@
+import math
+import functools
+
+
+
+infile = 'donors.txt'
+donor_objects = []
+
+
+def read_donors():
+    master_donor = []
+    
+    with open(infile) as donor_input:
+        donor_input = [w for line in open(infile) for w in line.rstrip('\n').split("\n")]
+        donor_input.remove("Donor Name - Donations with comma dilimiter")
+
+        for line in donor_input:
+            name_string = line.split('-')[0].strip()
+            donation_string = line.split('-')[1].strip(" ").strip()
+            donation_list = donation_string.split(',')
+
+            donation_list = [float(x) for x in donation_list]
+
+
+            temp = [name_string, donation_list]
+            master_donor.append(temp)
+            #donors[name_string.lower()] = [name_string, donation_list]
+    
+    
+    
+    # print(master_donor) 
+
+    for i, v in enumerate(master_donor):
+        # print(i)
+        # print(str(temp_name))
+        name = str(master_donor[i][0])
+        # print(name)
+        donation = master_donor[i][1]
+        # print(donation)
+        temp_name = Donor(name, donation)
+        # print(temp_name.name, temp_name.donation)
+        
+        donor_objects.append(temp_name)
+        DB.add_donor(temp_name)
+
+
+
+
+        # donation = list(master_donor[i][1])
+        # print(temp_name)
+        # temp_name = Donor(name, donation)
+        # print(temp_name.name)
+        # print(temp_name.donation)
+        # master_donor.append(temp_name)
+
+
+
+
+
+
+class Donor(object):
+    
+    # name = ""
+    # donations = []
+    def __init__ (self, name, donations):
+        
+        self.name = name
+        self.donations = donations
+
+    def add_donation(self, money):
+
+        self.donations.append(money)
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+    @property    
+    def maxdonation(self):
+        return max(self.donations)
+
+        
+
+    @property
+    def namelength(self):
+        return(len(self.name))
+
+    @property
+    def donation_count(self):
+        return(len(self.donations))
+
+    @property
+    def average_donation(self):
+        return(self.total/self.donation_count)
+
+
+
+
+
+
+class Storage:
+
+
+    def __init__(self):
+        self.donors = []
+
+
+    #need methods to add donors, add donation
+    # @classmethod
+    def add_donor(self, thing):
+        self.donors.append(thing)
+
+
+
+    def add_donation(self, name, new_donation):
```

Comment:
perhaps you should only use the add_donation() method of the donor object, rather than finding a donor and adding in one step.

in any case, you want a "find_a_donor" method in this class, which you could use if you want find and add at the same time.